### PR TITLE
refactor(frontend): Make dashboard search box the first filter

### PR DESCRIPTION
**Pull request by cemremengu**
### SUMMARY

Searching is the most natural filter so most people are looking for it first before checking others. This PR moves `Search` to the beginning of the dashboard filters in an attempt to provide easier accessibility.

Feel free to close this PR if this is not suitable for core team.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

** BEFORE **

image

**AFTER**

image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
@kasiazjc @jess-dillard any objections to this? If not, we can probably get it through quickly. @cemremengu (or anyone, really) would just need to resolve the conflicts

**Comment by jess-dillard** [comment]
@rusackas No objections here, let's do it!

**Comment by rusackas** [comment]
@cemremengu let me know if you can fix the merge conflict, and we'll get this thing through!

File: superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx [hunk 1]
```diff
@@ -533,13 +539,7 @@ function DashboardList(props: DashboardListProps) {
           { label: t('No'), value: false },
         ],
       },
-      {
-        Header: t('Search'),
         key: 'search',
```
Comment (rusackas) [inline superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx:542]: Oops! I think this line needed to come along for the ride!

**Review by rusackas — CHANGES_REQUESTED** [review]
One little tweak, then we should be good!

File: superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx (hunk 1 above)
Comment (cemremengu) [reply superset-frontend/src/views/CRUD/dashboard/DashboardList.tsx]: Ah apologies, forgot to check CI

**Review by rusackas — APPROVED** [review]
LGTM!

**Comment by sfirke** [comment]
Could this be done for Charts view also?  And I suppose Dataset view as well.

**Pull request merged**